### PR TITLE
Stop pending requests on disconnect

### PR DESCRIPTION
**Pull request by rgrinberg**
When we terminate a server, we need to make sure currently pending
requests are eliminated

### CODE REVIEW

**Review by ulugbekna — COMMENTED** [review]
Looks good. 

Notes:
1. I don't see code handling `exception Stopped of Message.request`. Who should be responsible for handling it? 
2. We could use some tests and logs, but that can happen in a later PR (perhaps by me)

**Comment by rgrinberg** [comment]
> I don't see code handling exception Stopped of Message.request. 

We don't handle this exception because our server does not make any requests to the client (at the moment).

> Who should be responsible for handling it?

The same we handle all errors that may be the result of making a request. The issue was that such errors would stall your fiber instead of raising an error. That makes debugging harder.

> We could use some tests and logs, but that can happen in a later PR (perhaps by me)

Feel free to add tests. It's a bit of an edge case so I don't feel it's very important, but it should be possible to create a test case that reflects the situation.

File: jsonrpc-fiber/src/jsonrpc_fiber.ml [hunk 1]
```diff
@@ -211,8 +227,16 @@ struct
...
+  let make_stop_pending_requests pending =
+    lazy
+      (let to_cancel = Table.fold pending ~init:[] ~f:(fun x acc -> x :: acc) in
+       Table.clear pending;
+       Fiber.parallel_iter to_cancel ~f:(fun ivar ->
+           Fiber.Ivar.fill ivar (Error `Stopped)))
+
   let create ?(on_request = on_request_fail)
       ?(on_notification = on_notification_fail) ~name chan state =
+    let pending = Table.create (module Id) 10 in
```
[the remark this answers is not in the document]
Comment (rgrinberg) [reply jsonrpc-fiber/src/jsonrpc_fiber.ml:239]: Good catch. Thanks.
